doc/rst/technotes/gpu.rst

```diff
@@ -136,39 +139,125 @@ GPU-Related Environment Variables
+  rewritten to call the cloned version of the outlined function when on a  GPU
+  locale. This mode increases overall code size but allows the compiler to
+  assume that a given function will execute on the GPU locale and optimize
+  accordingly. This may also be set by passing the ``chpl`` compiler the
+  ``--gpu-specialization`` flag. This is an experimental mode subject to change
+  in the future.
+* ``CHPL_GPU_NO_CPU_MODE_WARNING`` - this variable is relevant when using the
+  `CPU as Device mode`_ and if set causes it so that uses of
+  :proc:`~GPU.assertOnGpu` while in this mode will not generate a warning at
+  execution time. Alternatively, this can be done by passing
+  ``--gpuNoCpuModeWarning`` to your application. For more information, see the
+  `CPU as Device mode`_ section.
+
+Features
+--------------------
+
+In the following subsections we discuss various features of GPU supports.
 
-Chapel's build system will automatically try and deduce what type of GPU you
+Vendor Portability
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Chapel is able to generate code that will execute on either NVIDIA or AMD GPUs.
+Chapel’s build system will automatically try and deduce what type of GPU you
 have and where your installation of relevant runtime (e.g. CUDA or ROCm) are.
-If the type of GPU is not detected you may set ``CHPL_GPU`` manually to either
-``nvidia``, ``amd`` or ``cpu`` (See :ref:`below <cpu-mode-label>` for more
-information on this mode). If the relevant runtime path is not automatically
-detected (or you would like to use a different installation) you may set
-``CHPL_CUDA_PATH`` and/or ``CHPL_ROCM_PATH``.
-
-``CHPL_GPU_ARCH`` environment variable can be set to control the desired GPU
-architecture to compile for. For ``CHPL_GPU=nvidia``, this defaults to
-``sm_60``. For ``CHPL_GPU=amd`` it needs to be set explicitly. The environment
-setting can be overridden by the ``--gpu-arch`` compiler flag.  For a list of
-possible values please refer to "processor" values in `this table in the LLVM
-documentation <https://llvm.org/docs/AMDGPUUsage.html#processors>`_ for AMD or
-the `CUDA Programming Guide
+If the type of GPU is not detected you may set the ``CHPL_GPU`` environment
+variable manually to either ``nvidia`` or ``amd``.  ``CHPL_GPU`` may also
+manually be set to ``cpu`` to use `CPU as Device mode`_.
+
+Based on the value of ``CHPL_GPU``, Chapel's build system will also attempt to
+automatically detect the path to the relevant runtime. If it is not
+automatically detected (or you would like to use a different installation) you
+may set ``CHPL_CUDA_PATH`` and/or ``CHPL_ROCM_PATH`` explicitly.
+
+The CHPL_GPU_ARCH environment variable can be set to control the desired GPU
+architecture to compile for. The default value is ``sm_60`` for
+``CHPL_GPU_CODEGEN=cuda``. You may also use the ``--gpu-arch`` compiler flag to
+set GPU architecture. For a list of possible values please refer to "processor"
```

Comment:
It would be nice if you could formulate it such that "for AMD" came before the link; a reader like me might click the link before realizing the table isn't for CUDA.

Reply:
Modified to:

```
The ``CHPL_GPU_ARCH`` environment variable can be set to control the desired
GPU architecture to compile for. The default value is ``sm_60`` for
``CHPL_GPU_CODEGEN=cuda``. You may also use the ``--gpu-arch`` compiler flag to
set GPU architecture.  If using AMD, `this table in the LLVM documentation
<https://llvm.org/docs/AMDGPUUsage.html#processors>`_ has possible architecture
values (see the "processor" column). For NVIDIA, see the `CUDA Programming
Guide
<https://docs.nvidia.com/cuda/cuda-c-programming-guide/#features-and-technical-specifications>`_.
```